tmc-plugin/src/fi/helsinki/cs/tmc/ui/PreferencesUIFactory.java

```diff
@@ -76,24 +76,29 @@ public void showPreferencesDialog(final ActionListener dialogListener) {
         if (dialog != null) {
             throw new IllegalStateException("Preferences UI already visible");
         }
-        
+
         ActionListener closeListener = new ActionListener() {
             @Override
             public void actionPerformed(ActionEvent e) {
-                panel = null;
-                dialog = null;
+                destroyDialog();
                 dialogListener.actionPerformed(e);
             }
         };
-        
+
         DialogDescriptor descriptor = new DialogDescriptor(
                 panel,
                 "TMC Settings",
                 false,
                 NotifyDescriptor.OK_CANCEL_OPTION,
-                NotifyDescriptor.PLAIN_MESSAGE,
+                null,
```

Comment:
Why null parameter? I believe it might fix the bug but maybe there could be more abstraction/javadoc to explain this long function.